pkg/util/util.go

```diff
@@ -167,3 +168,20 @@ func GetAZFromTopology(topologyKey string, requirement *csi.TopologyRequirement)
 
 	return zone
 }
+
+func SanitizeLabel(input string) (string, error) {
+	// Replace non-alphanumeric characters (except '-', '_', '.') with '-'
+	reg := regexp.MustCompile(`[^-a-zA-Z0-9_.]+`)
+	sanitized := reg.ReplaceAllString(input, "-")
+
+	// Ensure the label starts and ends with an alphanumeric character
+	sanitized = strings.Trim(sanitized, "-_.")
+
+	// Ensure the label is not longer than 63 characters
+	if len(sanitized) > 63 {
+		sanitized = sanitized[:63]
+	}
+
+	// Convert to lowercase
+	return strings.ToLower(sanitized), nil
+}
```

Comment:
It doesn't seem this can ever return an error, so why bother with having `(string, error) as a return value?

Reply:
true, updated.